### ISSUE TITLE
Unable to Fetch Compare Product IDs

### ISSUE THREAD

**Issue by uandiweb** [scandipwa/scandipwa#2752]
Hi,

I just installed ScandiPWA on a Magento 2.4.2-p1 website: https://magento-577308-1980693.cloudwaysapps.com/

Every-time a page loads I got the message: "Unable to Fetch Compare Product IDs". See attachment below. 

Any idea what is wrong, and how to fix it?

The ScandiPWA information is: scandipwa/scandipwa 4.4.0, scandipwa-scripts 2.4.6, scandipwa/eslint-config 0.1.24, tilework/mosaic 0.1.1

Thank you!

<img width="373" alt="compare" src="https://user-images.githubusercontent.com/20729592/121528813-cf645280-c9c9-11eb-8dc4-8cf79e2de9a0.png">

**Comment by rMazeiks** [comment]
While I haven't reproduced this myself, there is [1 other Slack user that reported this error](https://scandipwa.slack.com/archives/CD8ETGY3Y/p1623333660210500) - perhaps there's a bug?

**Comment by websuits** [comment]
I have reproduced this on latest version.

**Comment by alipinsh** [comment]
I investigated this and found two temporary solutions:
1. Downgrade `scandipwa/compare-graphql` to 1.0.5
2. In your scandipwa folder add files from [this commit](https://github.com/scandipwa/scandipwa/commit/fe9f803052e36b7174020ff94cf6813564198976#diff-29c1e654c43a4526cd64d34a219a52f84dfe79519427f9280e8d528a5dc52b75). You also need to remove references to `updateCustomerIsAuthTokenExpired` in `MyAccount.dispatcher.js`, because it got removed in newer version.

**Comment by websuits** [comment]
@alipinsh  are any changes you made for your theme applied ? I found this issue while investigating the compare.

**Comment by alipinsh** [comment]
@websuits i just inserted the files from that commit i mentioned into my theme and also changed the `MyAccount.dispatcher.js` a bit because of the removed `updateCustomerIsAuthTokenExpired`. I checked on local, and it worked, but decided to remove all that and just downgrade the `scandipwa/compare-graphql` to the past version.

**Comment by websuits** [comment]
@alipinsh 

Yes, it works indeed. What i am asking is, do you have scandipwa installed as a Magento theme ? I was going to test your fix, installed it as a theme, and i noticed another issue, that when you make changes in src/ nothing changes in front-end. Might be another bug.

**Comment by alipinsh** [comment]
@websuits Well yes, it is installed as a magento theme. I think that is another issue what you are experiencing now. Is the theme linked? Have you run the `BUILD_MODE=magento npm run start` in scandipwa folder?

**Comment by websuits** [comment]
@alipinsh Yes, i did. Are you on slack?

**Comment by rajnibalayadavtest** [comment]
Can anyone tell me the downgrade mechanism of ```scandipwa/compare-graphql to 1.0.5``` ?

**Comment by rMazeiks** [comment]
This issue, while helpful to many, does not appear in the search results due to the screenshotted error message. Here is the text version, for search purposes:

```
[{"message":"Cannot query field \"compareProducts\" on type \"Query\". Did you mean \"products\"?","extensions":{"category":"graphql"},"locations":[{"line":1,"column":30}]}]
```

**Comment by carinadues** [comment]
This issue was solved in release 4.5.0.
